backend/oasst_backend/models/message_revision.py

```diff
@@ -0,0 +1,31 @@
+from datetime import datetime
+from typing import Optional
+from uuid import UUID, uuid4
+
+import sqlalchemy as sa
+import sqlalchemy.dialects.postgresql as pg
+from sqlmodel import Field, SQLModel
+
+from .payload_column_type import PayloadContainer, payload_column_type
+
+
+class MessageRevision(SQLModel, table=True):
+    __tablename__ = "message_revision"
+
+    id: UUID = Field(
+        sa_column=sa.Column(
+            pg.UUID(as_uuid=True), primary_key=True, default=uuid4, server_default=sa.text("gen_random_uuid()")
+        )
+    )
+
+    payload_type: str = Field(nullable=False, max_length=200)
+    payload: Optional[PayloadContainer] = Field(
+        sa_column=sa.Column(payload_column_type(PayloadContainer), nullable=True)
+    )
+
+    message_id: UUID = Field(sa_column=sa.Column(sa.ForeignKey("message.id"), nullable=False))
+    user_id: UUID = Field(sa_column=sa.Column(sa.ForeignKey("user.id"), nullable=False))
+    created_date: datetime = Field(
+        sa_column=sa.Column(sa.DateTime(timezone=True), nullable=False, server_default=sa.func.current_timestamp())
+    )
+    previous_revision_id: Optional[UUID] = Field(sa_column=sa.Column(sa.ForeignKey("message_revision.id")))
```

Comment:
could this be null if there are no previous revisions?

Reply:
Yes, a null previous revision would mean it is the original version of the message.

Reply:
Ok, I am just asking because it is a foreign key, so I though the database would complain if it was null.

Reply:
Should I make it not be a foreign key then? I'm not really sure I needed to in the first place.

Reply:
I've removed the `previous_revision_id` and `head_revision_id` fields since uuid7 can be used to sort based on time instead.